check_usage.py

```diff
@@ -0,0 +1,238 @@
+
+
+
+parser = argparse.ArgumentParser(description=docstr)
+
+parser.add_argument('-U', '--user',
+                                help='check usage of this user')
+parser.add_argument('-A', '--account',
+                                help='check usage of this account')
+
+parser.add_argument('-s', '--start', type=valid_date,
+                    help='start time YYYY-MM-DD[THH:MM:SS]'
+                    '(DEFAULT: {}-06-01T00:00:00)'.format(datetime.datetime
+                                                          .now().year),
+                    default='{}-06-01T00:00:00'.format(datetime.datetime
+                                                       .now().year))
+parser.add_argument('-e', '--end', type=valid_date,
+                    help='end time YYYY-MM-DD[THH:MM:SS]'
+                    '(DEFAULT: Current TimeStamp)',
+                    default=datetime.datetime.now()
+                    .strftime(timestamp_format_complete))
+parsed = parser.parse_args()
+user = parsed.user
+account = parsed.account
+_start = parsed.start
+_end = parsed.end
+start = process_date_time(_start)
+end = process_date_time(_end)
+
+if not (user or account):
+    parser.error('Atleast one of -U/--user and -A/--account required.')
+
+
+
+##### prepare params #####
+
+
+
+# base_url = 'https://scgup-dev.lbl.gov:8443/mybrc-rest'
+base_url = 'http://localhost:8880/mybrc-rest'
+request_urls = {}
+output_headers = {}
+
+if user:
+    output_header = 'Usage for USER {} [{}, {}]: '.format(user, _start, _end)
+    request_params = {
+        'start_time': start,
+        'end_time': end,
+        'user': user
+    }
+    url_usages = base_url + '/user_account_usages?' + \
+        urllib.urlencode(request_params)
+
+    request_urls['user'] = url_usages
+    output_headers['user'] = output_header
+
+if account:
+    output_header = 'Usage for ACCOUNT {} [{}, {}]: '.format(account, _start, _end)
+    request_params = {
+        'start_time': start,
+        'end_time': end,
+        'account': account
+    }
+    url_usages = base_url + '/account_usages?' + \
+        urllib.urlencode(request_params)
+
+    request_urls['account'] = url_usages
+    output_headers['account'] = output_header
+
+if user and account:
+    output_header = 'Usage for USER:ACCOUNT {}:{} [{}, {}]: '.format(user, account, _start ,_end)
+    request_params = {
+        'start_time': start,
+        'end_time': end,
+        'user': user,
+        'account': account
+    }
+    url_usages = base_url + '/user_account_usages?' + \
+        urllib.urlencode(request_params)
+
+    request_urls['user_account'] = url_usages
+    output_headers['user_account'] = output_header
+
+
+
+##### output functions #####
+
+
+
+def get_allocation_for_account(account):
+    request_params = {'name': account}
+    req_url = base_url + '/projects?' + \
+        urllib.urlencode(request_params)
+
+    try:
+        req = urllib2.Request(req_url)
+        req.add_header("Authorization",
```

Comment:
We can remove references to tokens, since this script is only making `GET` requests.

Reply:
yeah might as well... done! Thanks :)